hudi-spark/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -108,11 +109,13 @@ private[hudi] object HoodieSparkSqlWriter {
       handleSaveModes(mode, basePath, tableConfig, tblName, operation, fs)
       // Create the table if not present
       if (!tableExists) {
-        val archiveLogFolder = parameters.getOrElse(
-          HoodieTableConfig.HOODIE_ARCHIVELOG_FOLDER_PROP_NAME, "archived")
+        val archiveLogFolder = parameters.get(HoodieTableConfig.HOODIE_ARCHIVELOG_FOLDER_PROP_NAME).get
+        val hoodieWriteConfig = DataSourceUtils.createHoodieConfig(Schema.create(Schema.Type.NULL).toString, path.get, tblName, mapAsJavaMap(parameters))
+        val index = SparkHoodieIndex.createIndex(hoodieWriteConfig)
```

Comment:
this is problematic. We are initializing an `index`, just for this. We should refrain from accessing anything beyond WriteClient at this level.

Reply:
you can just convert the config value to the enum right?

Reply:
If your goal was to just validate by turning the config to enum to string, then can we validate at the WriteConfig builder level? That will leave this code very clean.

Reply:
done, just get index from HoodieIndexUtils. getIndexType , which get from writeconfig